### ISSUE TITLE
RE2.getUtf16Length crashes Node when it encounters a partial or invalid sequence at the end of a buffer

### ISSUE THREAD

**Issue by Exter-N** [uhop/node-re2#31]
When `RE2.getUtf16Length` encounters a partial or invalid UTF-8 sequence at the end of a buffer, it crashes Node (as of v8.2.1), for example :
```js
RE2.getUtf16Length(new Buffer([ 0xF0 ]));
```
Gives the following :
```
Segmentation fault (core dumped)
```

**Comment by uhop** [comment]
Again, it is clearly a bug of the GIGO variety.

**Comment by Exter-N** [comment]
While in the other cases the "Garbage In" has to explicitly come from the developer which consumes this library, in this one it can easily be read from a corrupt file or the network with insufficient validation, which makes it more severe.

**Comment by Exter-N** [comment]
This one should have been fixed by #32.
